scripts/ci/selective_tests.sh

```diff
@@ -125,18 +182,60 @@ function count_changed_files() {
     echo "${count_changed_files}"
 }
 
-function run_all_tests_when_push_or_schedule() {
+function run_everything_when_push_or_schedule() {
     if [[ ${GITHUB_EVENT_NAME} == "push" || ${GITHUB_EVENT_NAME} == "schedule" ]]; then
         echo
         echo "Always run all tests in case of push/schedule events"
         echo
-        set_outputs_run_all_tests
-        exit
+        set_outputs_run_everything_and_exit
+    fi
+}
+
+function check_if_api_tests_should_be_run() {
+    local api_triggering_patterns=(
+        "^airflow/api"
+    )
+    pattern_array=("${api_triggering_patterns[@]}")
+    show_changed_files "$(get_regexp_from_patterns)"
+
+    if [[ $(count_changed_files) == "0" ]]; then
+        does_not_need_api_tests
+    else
+        needs_api_tests
+    fi
+}
+
+function check_if_helm_tests_should_be_run() {
+    local helm_triggering_patterns=(
+        "^chart"
+    )
+    pattern_array=("${helm_triggering_patterns[@]}")
+    show_changed_files "$(get_regexp_from_patterns)"
+
+    if [[ $(count_changed_files) == "0" ]]; then
+        does_not_need_helm_tests
+    else
+        needs_helm_tests
+    fi
+}
+
+function check_if_docs_should_be_generated() {
+    local docs_triggering_patterns=(
+        "^docs$"
+        "\.py$"
+    )
+    pattern_array=("${docs_triggering_patterns[@]}")
+    show_changed_files "$(get_regexp_from_patterns)"
+
+    if [[ $(count_changed_files) == "0" ]]; then
+        echo "None of the docs changed"
+    else
+        image_build_needed="true"
     fi
 }
 
 function check_if_tests_should_be_run_at_all() {
-    TEST_TRIGGERING_PATTERNS=(
+    local api_triggering_patterns=(
```

Comment:
Could this fn not be reimplemented instead to just check that none of the previous selective tests ran anything? Saves having to keep multiple things in sync.

Reply:
Not really, because we have to handle the logic to check if any "other" files changed. This is especially important to account for future changes - in order to be on the safe side I count the things that i KNOW can be safely tested in isolation and everything else is considered as ("trigger all tests" if they change)